src/query/storages/common/index/src/bloom_index.rs

```diff
@@ -395,7 +395,7 @@ impl BloomIndex {
         )?;
 
         let (new_expr, _) =
-            ConstantFolder::fold_with_domain(&expr, &domains, &self.func_ctx, &BUILTIN_FUNCTIONS);
+            ConstantFolder::fold_for_prune(&expr, &domains, &self.func_ctx, &BUILTIN_FUNCTIONS);
```

Comment:
It looks like `fold_for_prune` is meant to rewrite the equal expression with a cast column function, like `col::dest_type = val` as `col = val::src_type`, which we can rewrite directly inside the `visit_expr_column_eq_constant` function and don't need redefine `fold_for_prune` function. If `val::src_type` fails, the result must be false.